R/combine_GVA.R

```diff
@@ -69,123 +71,70 @@ combine_GVA <- function(
-  # `BB16_GVA` (column Q in `Working File` worksheet).
-
-  GVA_sectors <- dplyr::left_join(GVA_sectors, GVA, by = c('SIC2' = 'SIC', 'year'))
-  GVA_sectors$BB16_GVA <-  GVA_sectors$perc_split * GVA_sectors$GVA
-
-  ### 4. Calculate sums across sectors and years ----
-
-  # Finally calculate the sums across all sectors and years.
-
-  GVA_by_sector <- dplyr::group_by_(GVA_sectors, ~year, ~sector)
-  GVA_by_sector <- dplyr::summarise_(GVA_by_sector, GVA = ~sum(BB16_GVA))
-
-  #### 5. Add in total UK GVA from gva ----
-
-
-  # Use create a table to merge in with GVA_by_sector
-
-  GVA_UK <- GVA[which(grepl('total.*intermediate.*',GVA$SIC)),]
-  GVA_UK$sector <- 'UK'
-  GVA_UK <- GVA_UK[,c('year', 'sector', 'GVA')]
-
-  # Merge this into `GVA_by_sector` by full join
-
-  GVA_by_sector <- dplyr::full_join(GVA_by_sector, GVA_UK)
-
-  ### 6. Match in Tourism data ----
-
-  # Tourism data is provided in a separate spreadsheet and imported in the
-  # `tourism` object
-
-  tourism_UK <- tourism
-  tourism_UK$sector <- 'tourism'
-  tourism_UK <- tourism_UK[, c('year', 'sector', 'GVA')]
-  GVA_by_sector <- dplyr::full_join(tourism_UK, GVA_by_sector)
-
-  ### 7. Add tourism overlap ----
-
-  # Also need to add the `$overlap` from tourism to the `all_dcms` totals in
-  # `GVA_by_sector`
-
-  tourism_all_sectors <- tourism
-  tourism_all_sectors$sector <- 'all_dcms'
-  tourism_all_sectors <- tourism_all_sectors[ ,c('year', 'sector', 'overlap')]
-  GVA_by_sector <- dplyr::right_join(tourism_all_sectors, GVA_by_sector)
-  GVA_by_sector$GVA <- ifelse(
-    !is.na(GVA_by_sector$overlap),
-    GVA_by_sector$overlap + GVA_by_sector$GVA,
-    GVA_by_sector$GVA
-  )
-  GVA_by_sector <- GVA_by_sector[, !names(GVA_by_sector) == 'overlap']
-
-  #### 8. Build the GVA_by_sector dataframe expected by the eesectors package ----
-
-  # Build df to match eesectors::GVA_by_sector_2016
-  # Will need adjusting in future versions
-
-  GVA_by_sector <- GVA_by_sector[GVA_by_sector$year %in% 2010:2015,]
-  GVA_by_sector <- dplyr::ungroup(GVA_by_sector) # May not be required
-  GVA_by_sector$GVA <- round(GVA_by_sector$GVA, 2)
-  GVA_by_sector$sector <- factor(GVA_by_sector$sector)
-  GVA_by_sector$year <- as.integer(GVA_by_sector$year)
-  GVA_by_sector <- GVA_by_sector[, c('sector', 'year', 'GVA')]
-  GVA_by_sector <- dplyr::arrange_(GVA_by_sector, ~year, ~sector)
 
-  return(GVA_by_sector)
+  #Annual business survey, duplicate 2014 data for 2015 and
+  #then duplicate non SIC91 then add SIC 91 with sales data
+  ABS_2015 <- filter(ABS, year == 2014) %>%
+    mutate(year = 2015) %>%
+
+    #this line makes no sense to me - we are just duplicated rows we already
+    #have so surely it is redundant??
+    bind_rows(filter(ABS, !SIC %in% unique(SIC91$SIC))) %>%
+    bind_rows(SIC91)
+
+
+  # keep cases from ABS which have integer SIC - which is just a higher level SIC
+  denom <- filter(ABS_2015, SIC %in% unique(eesectors::DCMS_sectors$SIC2)) %>%
```

Comment:
If you don't import dplyr as suggested, you would need to do `importFrom(dplyr, '%>%')` to continue to use the pipe. Also note that dplyr can pass data into the first argument, so that: `denom <- filter(ABS_2015, SIC %in% unique(eesectors::DCMS_sectors$SIC2))` becomes `denom <- ABS_2015 %>% filter(SIC %in% unique(eesectors::DCMS_sectors$SIC2))`.